syntax = "proto3";

option java_package = "com.bookmyshow.proto";
option java_outer_classname = "MovieProto";

message MovieSuccessResponse {
  string message = 1;
  int32 status = 2;
  string movieId = 3;
}

message AllMoviesResponse {
  repeated Movie movies = 3;
}

message Movie {
  string movieId = 1;
  string title = 2;
  string description = 3;
  repeated string genre = 4;
  int32 durationMins = 5;
  string releaseDate = 6;
  double rating = 7;
  repeated string languages = 8;
  string imageKey = 9;
  string posterKey = 10;
}

message MovieInput {
  string title = 1;
  string description = 2;
  repeated string genre = 3;
  int32 durationMins = 4;
  string releaseDate = 5;
  double rating = 6;
  repeated string languages = 7;
  string imageKey = 8;
  string posterKey = 9;
}

message MoviesList {
  repeated Movie movies = 1;
}
